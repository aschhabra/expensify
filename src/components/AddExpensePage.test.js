import React from "react";
import {shallow} from "enzyme";
import {AddExpensePage} from "./AddExpensePage";
import testexpenses from "../fixtures/expenses"


let onSubmit,history, wrapper;
beforeEach(() => {
  onSubmit = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<AddExpensePage onSubmit={onSubmit} history={history} />);
});


test("should render AddExpensepage Correctly",()=>{

  expect(wrapper).toMatchSnapshot();
  

})

 test("shoudl handle onSubmit",()=>{
   wrapper.find("ExpenseForm").prop("onSubmit")(testexpenses[0]);
   expect(history.push).toHaveBeenLastCalledWith("/");
   expect(onSubmit).toHaveBeenLastCalledWith(testexpenses[0]);
   
 
 })
