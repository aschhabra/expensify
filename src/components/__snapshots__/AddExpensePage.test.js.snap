// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddExpensepage Correctly 1`] = `
<div>
  This is from add expense
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
